pkg/backup/safe_point.go

```diff
@@ -52,3 +54,32 @@ func UpdateServiceSafePoint(ctx context.Context, pdClient pd.Client, ttl int64,
 		brServiceSafePointID, ttl, backupTS-1)
 	return err
 }
+
+// StartServiceSafePointKeeper will run UpdateServiceSafePoint periodicity
+// hence keeping service safepoint won't lose.
+func StartServiceSafePointKeeper(
+	ctx context.Context,
+	ttl int64,
+	pdClient pd.Client,
+	backupTS uint64,
+) {
+	// At least 1 second gap, or time.NewTicker will blame us.
+	gapSec := time.Duration(utils.MaxInt(int(ttl/5), 1))
```

Comment:
what's special about "5"?

btw please make this a true duration:

```go
gap := time.Duration(ttl) * time.Second / 5
if gap < time.Second {
    gap = time.Second
}
```

Reply:
The default ttl is 5 mins... So I just choice one of its factor.

Do we need so frequently reset it? The origin version reset it every time a range is fully backed up.

Reply:
(And every 5 seconds...)

Reply:
As long as it is reset before the TTL it is fine